libraries/AR_Motors/AP_MotorsUGV.cpp

```diff
@@ -214,7 +226,58 @@ void AP_MotorsUGV::set_throttle(float throttle)
     }
 
     // check throttle is between -_throttle_max and  +_throttle_max
-    _throttle = constrain_float(throttle, -_throttle_max, _throttle_max);
+    float throttle_chk = constrain_float(throttle, -_throttle_max, _throttle_max);
+    _esc_backward_unlock = constrain_int16(_esc_backward_unlock,0,1);
+    if ((throttle_chk < 0) && (_esc_backward_unlock != 0))
+    {
+        if (_esc_backward_unlock_state ==0)
```

Comment:
Not really a fan of all these else ifs. I think a switch case would be better, with a enum for the state. You could move to time based stepping so it will be unaffected by the call rate.

Reply:
good idea... 
done.